2005-01-05  Sergey Poznyakoff  <gray@Mirddin.farlep.net>

	* src/tar.c (parse_opt): Bugfix: Use ARGP_KEY_ARG. Thanks
	Mike Frysinger <vapier@gentoo.org> for reporting.

--- src/tar.c	22 Dec 2004 10:53:48 -0000	1.109
+++ src/tar.c	5 Jan 2005 17:30:36 -0000	1.110
@@ -617,9 +617,9 @@ parse_opt(int key, char *arg, struct arg
   
   switch (key)
     {
-      case 1:
+      case ARGP_KEY_ARG:
 	/* File name or non-parsed option, because of ARGP_IN_ORDER */
-	name_add (optarg);
+	name_add (arg);
 	args->input_files++;
 	break;
 
